pkg/operation/operation.go

```diff
@@ -650,3 +649,32 @@ func (o *Operation) ToAdvertisedAddresses() []gardencorev1beta1.ShootAdvertisedA
 
 	return addresses
 }
+
+// StoreCheckSum stores the passed checksum under the given key from the operation. Calling this function is thread-safe.
+func (o *Operation) StoreCheckSum(key, value string) {
+	o.checkSumsMutex.Lock()
+	defer o.checkSumsMutex.Unlock()
+
+	if o.checkSums == nil {
+		o.checkSums = make(map[string]string)
+	}
+
+	o.checkSums[key] = value
+}
+
+// LoadCheckSum loads the checksum value under the given key from the operation. Calling this function is thread-safe.
+func (o *Operation) LoadCheckSum(key string) string {
```

Comment:
I am thinking if this function now have to handle the case when the key does not exist?
Currently it simply returns empty string.

Reply:
You're right, this is the more secure way. However, we already read values from the `CheckSum` map directly before, see this [example](https://github.com/gardener/gardener/blob/543217d363c57073e0bf681f19937e95fd6419c9/pkg/operation/botanist/controlplane.go#L65). I decided to adopt this 1:1 to only address the concurrency problem in this PR.

Reply:
Thanks for the explanation!